File: swift/singleNumber.swift
func singleNumber(nums: [Int]) -> Int {}
    let sortedNums = nums.sort()
    var result = sortedNums.last!
    for i in 0..<sortedNums.count {
        if i % 2 == 0 && i + 1 < sortedNums.count {
            if sortedNums[i] != sortedNums[i+1] {
                result = sortedNums[i]
                break
            }
        }
    }
    return result
}

File: ruby/single_number.rb
def single_number(nums)
   nums.sort!
   result = nums.last
   nums.each_with_index do |num, index|
       if index % 2 == 0 and index + 1 < nums.count
           if num != nums[index+1]
               result = num
               break
           end
       end
   end
   return result
end
